File: examples/array_reverse.pl
use strict;
use warnings;
use FFI::Platypus 2.00;

my $ffi = FFI::Platypus->new(
  api => 1,
  lib => './array_reverse.so',
);

$ffi->attach( array_reverse   => ['int[]','int'] );
$ffi->attach( array_reverse10 => ['int[10]'] );

my @a = (1..10);
array_reverse10( \@a );
print "$_ " for @a;
print "\n";

@a = (1..20);
array_reverse( \@a, 20 );
print "$_ " for @a;
print "\n";

File: examples/array_reverse.c
void
array_reverse(int a[], int len) {
  int tmp, i;

  for(i=0; i < len/2; i++) {
    tmp = a[i];
    a[i] = a[len-i-1];
    a[len-i-1] = tmp;
  }
}

void
array_reverse10(int a[10]) {
  array_reverse(a, 10);
}
